scripts/ci/check_compliance.py

```diff
@@ -1060,6 +1060,28 @@ def run(self):
                 self.failure(failure)
 
 
+class BinaryFiles(ComplianceTest):
+    """
+    Check that the diff contains no binary files.
+    """
+    name = "BinaryFiles"
+    doc = "No binary files allowed."
+    path_hint = "<git-top>"
+
+    def run(self):
+        BINARY_ALLOW_PATHS = ("doc/", "boards/")
+        BINARY_ALLOW_EXT = ("jpg", "jpeg", "png", "svg")
```

Comment:
`svg` is not a binary

Reply:
I know but somehow git detects some of them as binary, not too sure why, this is without it:

```
ERROR   : Test BinaryFiles failed: 
Binary file not allowed: boards/posix/doc/Zephyr_and_bsim.svg
ERROR   : Test BinaryFiles failed: 
Binary file not allowed: boards/posix/doc/layering.svg
ERROR   : Test BinaryFiles failed: 
Binary file not allowed: doc/_static/images/logo-readme-dark.svg
ERROR   : Test BinaryFiles failed: 
Binary file not allowed: doc/services/zbus/images/zbus_anatomy.svg
ERROR   : Test BinaryFiles failed: 
Binary file not allowed: doc/services/zbus/images/zbus_operations.svg
ERROR   : Test BinaryFiles failed: 
Binary file not allowed: doc/services/zbus/images/zbus_overview.svg
```

Not too sure why, I'll dig a bit.

Reply:
This is weird:

```
$ echo test > test
$ echo test > test.svg
$ git add test test.svg 
$ git diff --cached
diff --git a/test b/test
new file mode 100644
index 0000000000..9daeafb986
--- /dev/null
+++ b/test
@@ -0,0 +1 @@
+test
diff --git a/test.svg b/test.svg
new file mode 100644
index 0000000000..9daeafb986
Binary files /dev/null and b/test.svg differ
```

Aha!

https://github.com/zephyrproject-rtos/zephyr/blob/b38eb53e003bee6afa377dce50a71b75287b2647/.gitattributes#L7-L8

This gave me a bit of a headscratching.  Let me add a comment about it.

Reply:
I actually added that because many SVGs come from GUI apps (Inkscape, Visio, etc.) so when you edit them the diff is useless.

Reply:
Ah happy enough, just means this check has to list the extension (or somehow skip gitattributes but  I think listing it is fine)